pos_partner_firstname/static/src/js/ClientDetailsEdit.js

```diff
@@ -0,0 +1,101 @@
+odoo.define("pos_partner_firstname.ClientDetailsEdit", function (require) {
+    "use strict";
+
+    const {_t} = require("web.core");
+    const ClientDetailsEdit = require("point_of_sale.ClientDetailsEdit");
+    const Registries = require("point_of_sale.Registries");
+
+    const PosClientDetailsEdit = (ClientDetailsEdit) =>
+        class extends ClientDetailsEdit {
+            constructor() {
+                super(...arguments);
+                var self = this;
+                this.partner_names_order = "last_first";
+                this.rpc({
+                    model: "res.partner",
+                    method: "get_names_order",
+                    args: [],
+                }).then(function (partner_names_order) {
+                    if (partner_names_order) {
+                        self.partner_names_order = partner_names_order;
+                    }
+                });
```

Comment:
```suggestion
                this
                .rpc({model: "res.partner", method: "get_names_order", args: []})
                .then((res) => this.partner_names_order = res || "last_first");
```

No need to use `self = this` in modern js, you can just use arrow functions for callbacks :)